pkg/async/pool/pool.go

```diff
@@ -228,9 +226,7 @@ func (p *Pool) worker(ctx context.Context) {
 
 // Close blocks until all workers finshes current items and terminates
 func (p *Pool) Close() {
-	p.contextMu.Lock()
-	p.cancel(&orerr.ShutdownError{Err: context.Canceled})
-	p.contextMu.Unlock()
+	close(p.closed)
```

Comment:
your comment makes me think - we should make this safer regardless:
```suggestion
	if p.closed != nil {
		close(p.closed)
	}
```